### PR TITLE
Add `labs/` folder

### PR DESCRIPTION
**Pull request by ShadowJonathan**
This adds a `labs` folder for experimental crates, code, and other misc items.

### CODE REVIEW

**Comment by gnunicorn** [comment]
Thanks, @ShadowJonathan !

To give some more context, I would like to elaborate a bit more our thinking behind this:

As some of you might have noticed, we are branching out with the rust-sdk into user territory on various fronts right now; via FFI to Javascript, Swift for iOS and Android. While _we_ love Rust, we also know that many will be using our SDK through higher level abstractions, potentially in other language or build on top of this SDK.

To ensure they receive a good quality and we have control over their experience, we want to have as many things as feasible checked for with our CI before merging PRs. At the same time, as we are branching out, we are also aware that the APIs need to evolve and change, concepts might be tried out and thrown away and specific things might need a few iterations before they are mature enough for us to committing to longer term support for them. This is what this section in this repo is going to be about.

In `labs` you'll find new and novel approaches, partial implementations and experiments to support specific use cases or platforms. As the Readme states, we are not providing the same long-term-support for these, they might be subject to change or development might stop at any time without further notice. They might also contain multiple different approaches to the same problem. That does not mean we might not also publish these to `crates.io` (though always with the note about their experimental nature).

File: labs/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+`matrix-sdk-crypto`, etc.).
+
+Code and documentation here may become unmaintained or outdated, and possibly removed at any time.
+
+---
+
+That said, this directory is meant to freely explore unconventional or interesting ways the Matrix
+Rust SDK can evolve, feel free to propose an experiment.
+
+Current experiments;
```
Comment (gnunicorn) [inline labs/README.md:14]:
```suggestion
## Current experiments
```

File: labs/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+TBD
```
Comment (gnunicorn) [inline labs/README.md:16]:
```suggestion
TBD


## Archived experiments

_Link to PR that deleted the experiment from the repo, newest first_: 
```

File: labs/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Experiments
+
+This directory contains experiments, work-in-progress crates, or other code and documentation, that
+do not fall under the same stability guarantees as the main crates (`matrix-sdk`,
+`matrix-sdk-crypto`, etc.).
+
+Code and documentation here may become unmaintained or outdated, and possibly removed at any time.
```
Comment (gnunicorn) [inline labs/README.md:7]: I do want project that are here to be maintained, at least in the sense that CI-jobs ensure we don't break them unknowingly. But with a breaking CI we might also just agree that an experiment has run its course and be removed. So, let's keep this a bit more vague...

```suggestion
Lab projects might be abandoned and possibly removed at any time.
```

**Review by gnunicorn — APPROVED** [review]
Some suggestions.

**Review by gnunicorn — APPROVED** [review]

**Review by poljar — APPROVED** [review]

**Pull request merged**